python/activator/middleware_interface.py

```diff
@@ -327,12 +321,15 @@ def __init__(self, central_butler: Butler, image_bucket: str, visit: FannedOutVi
                  skymap: str, local_repo: str, local_cache: DatasetCache,
                  prefix: str = "s3://"):
         self.visit = visit
+        # Usually prompt processing only cares about on-sky images and expects all of them
+        # to have equatorial coordinates and sky rotation angles. However here we only warn
+        # and not raise so to allow special cases such as using calibration frames to test.
         if self.visit.coordinateSystem != FannedOutVisit.CoordSys.ICRS:
-            raise ValueError("Only ICRS coordinates are supported in Visit, "
-                             f"got {self.visit.coordinateSystem!r} instead.")
+            _log.warning("Got non-ICRS coordinates %s in Visit %s.",
+                         repr(self.visit.coordinateSystem), self.visit)
```

Comment:
```suggestion
            _log.warning("Got non-ICRS coordinates %r in Visit %s.",
                         self.visit.coordinateSystem, self.visit)
```

Reply:
Won't this still break when trying to preload spatial datasets (refcats and templates)?

Reply:
The nextVisit of those calibration frames have [ra,dec]=[0,0], so they don't "break" but just waste time copying non-useful data.  But it's true that we may not rely on them always being [0,0] as the default; we might tell summit to do so assuming we can have our own special "prompt-processing-test" observing script.

Reply:
I'm confused. What is the resolution of this issue?

Reply:
(Sorry. I did not remember hitting "Resolve conversation" here and must have been a mistake)

Reply:
Coming back to this after a long time... 

A bad value for `dec` indeed would break `prep_butler`, but any numeric values for `ra` and `cameraAngle` would be fine; any processed results after ISR would be suspicious of course.  I changed it so it still raises if values are obviously bad, otherwise just logs a message and continue.  

For the default case of [ra,dec]=[0,0], I think wasting time downloading potentially useless refcats is not a big deal. 

What do you think?

Reply:
I'd prefer to make loading of spatial datasets be conditional on having spatial coordinates in the first place, just because it's less error-prone than something that just happens to work. But I agree that run time is not a problem.

Reply:
I hesitated on making these visits handled any differently and found the happenstance convenient, but you have a good point. It's indeed more robust to not use those coordinates at all.  The last commit has been rewritten, and now the pointless loading will be skipped for non-on-sky visits.